src/sentry/api/endpoints/project_details.py

```diff
@@ -610,7 +614,10 @@ def put(self, request: Request, project) -> Response:
 
         if "dynamicSampling" in result:
             raw_dynamic_sampling = result["dynamicSampling"]
-            fixed_rules = self._fix_rule_ids(project, raw_dynamic_sampling)
+            try:
+                fixed_rules = self._fix_rule_ids(project, raw_dynamic_sampling)
+            except UniformDynamicSamplingRuleException as e:
+                return Response({"detail": str(e)}, status=400)
```

Comment:
I don't think that's the right way to do it.
Why do we mix permissions with validation?

We should handle invalid content by sending validation errors (which are a standard Django Rest Framework thing)